### PR TITLE
feat: simplify parameter declaration

### PR DESCRIPTION
**Pull request by wep21**
@clydemcqueen @jbohren I added some modification to simplify parameter declaration and fixed some lints. Could you review this?

### CODE REVIEW

**Comment by wep21** [comment]
@jbohren I changed the license from BSD to Apache 2 because Apache 2 is recommended for ros2. If there is any issue, I will revert them.

**Comment by jbohren** [comment]
> @jbohren I changed the license from BSD to Apache 2 because Apache 2 is recommended for ros2. If there is any issue, I will revert them.

Yeah that's good, though gstreamer is LGPL, with plugins that come with a variety of licenses though, so it might be worth drawing attention to that.

File: src/gscam.cpp [hunk 1]
```diff
@@ -103,11 +95,9 @@ namespace gscam {
     }
 
     // Get TF Frame
-    declare_parameter("frame_id", rclcpp::ParameterType::PARAMETER_STRING);
-    if(!get_parameter("frame_id",frame_id_)){
-      frame_id_ = "/camera_frame";
+    frame_id_ = declare_parameter("frame_id", "camera_frame");
```
Comment (clydemcqueen) [inline src/gscam.cpp]: The default value is a change from "/camera_frame"  but "camera_frame" is clearly better. I suspect that most ROS1 users of gscam will override the default anyway, so porting shouldn't be a problem. I'm dropping a comment here just to note the decision.

File: src/gscam.cpp [hunk 2]
```diff
@@ -45,12 +45,11 @@ namespace gscam {
   {
     // Get gstreamer configuration
     // (either from environment variable or ROS param)
-    std::string gsconfig_rosparam = "";
     bool gsconfig_rosparam_defined = false;
     char *gsconfig_env = NULL;
 
-    declare_parameter("gscam_config", rclcpp::ParameterType::PARAMETER_STRING);
-    gsconfig_rosparam_defined = get_parameter("gscam_config",gsconfig_rosparam);
+    const auto gsconfig_rosparam = declare_parameter("gscam_config", "");
```
Comment (clydemcqueen) [inline src/gscam.cpp]: Ah, nice! This is simpler, and it compiles w/o warnings on Foxy, Galactic and Rolling.

**Comment by clydemcqueen** [comment]
@jbohren thank you for jumping in on this.

@wep21 very nice improvements and LGTM, but I noticed that the README refers to *.launch instead of *.launch.xml. I should have caught this earlier. Can you update the README?

Update: I'll merge this & the edit the README to keep things moving. Thanks again.

**Review by clydemcqueen — APPROVED** [review]

**Pull request merged**